### ISSUE TITLE
MacOS Static Libs

### ISSUE THREAD

**Issue by tlightsky** [getnamo/socketio-client-ue4#15]
Is there a macos version

**Comment by getnamo** [comment]
Not atm, but @deams51 is working on a linux compile; with both systems based on unix a port to mac then should be reasonable.

The https://github.com/getnamo/socketio-client-prebuild static libraries just need to be recompiled for mac osx and included in the build.cs, then the rest should just work in theory. If you can get those compiled for osx, make a pull request!

**Comment by tlightsky** [comment]
I get macos run successfully
how should I upload these prebuid .a and build script?

**Comment by getnamo** [comment]
Great work! Make a pull request for the build script on: https://github.com/getnamo/socketio-client-prebuild and add the .a's directly on this repository as a pull request. I'll merge them in and the plugin will have your updates going forward :)

**Comment by tlightsky** [comment]
cause my submodule clone through https is not very good, I use 1.60.0 tar source,
I think it didn't affect the result.
I'll figure out there two pull request maybe next monday.

**Comment by getnamo** [comment]
merged the macos into master, can you build the latest version of the plugin and let me know if it all works as usual?

**Comment by getnamo** [comment]
merged a pull request (https://github.com/getnamo/socketio-client-ue4/pull/61) from @skatpgusskat with updated prebuilt libraries for macos. I couldn't personally verify, but if any mac user can sync to latest master and test it out, It would be great to get a second confirmation everything works!

**Comment by anadin** [comment]
unfortunately this does not appear to be the fix :(

output belopw
`Setting up Mono
/Users/Shared/Epic Games/UE_4.19/Engine /Users/Shared/Epic Games/UE_4.19/Engine/Binaries/Mac
Using 'git status' to determine working set for adaptive non-unity build.
WARNING: Failed to read makefile: Object reference not set to an instance of an object
Creating makefile for SocketIO (couldn't read existing makefile)
@progress push 5%
@progress pop
Performing 1 actions (8 in parallel)
[1/1] Link UE4Editor-SocketIOClient.dylib
ld: warning: object file (/Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/ThirdParty/SocketIO/Lib/Mac/libsioclient.a(sio_client.cpp.o)) was built for newer OSX version (10.12) than being linked (10.11)
ld: warning: object file (/Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/ThirdParty/SocketIO/Lib/Mac/libsioclient.a(sio_socket.cpp.o)) was built for newer OSX version (10.12) than being linked (10.11)
ld: warning: object file (/Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/ThirdParty/SocketIO/Lib/Mac/libsioclient.a(sio_client_impl.cpp.o)) was built for newer OSX version (10.12) than being linked (10.11)
ld: warning: object file (/Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/ThirdParty/SocketIO/Lib/Mac/libsioclient.a(sio_packet.cpp.o)) was built for newer OSX version (10.12) than being linked (10.11)
ld: warning: object file (/Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/ThirdParty/Boost/Lib/Mac/libboost_system.a(error_code.o)) was built for newer OSX version (10.12) than being linked (10.11)
Undefined symbols for architecture x86_64:
  "sio::client::connect(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > const&)", referenced from:
      UE4Function_Private::TFunctionRefCaller<FSocketIONative::Connect(FString const&, TSharedPtr<FJsonObject, (ESPMode)0> const&, TSharedPtr<FJsonObject, (ESPMode)0> const&)::$_10, void ()>::Call(void*) in Module.SocketIOClient.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ERROR: UBT ERROR: Failed to produce item: /Users/anadin/Dropbox/Projects/StealthStudios/UE4/SocketIO/Plugins/socketio-client-ue4/Binaries/Mac/UE4Editor-SocketIOClient.dylib
Total build time: 12.06 seconds (Local executor: 0.00 seconds)
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Experimental/AlembicImporter/Binaries/Mac/UE4Editor-AlembicImporter.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Experimental/AlembicImporter/Binaries/Mac/UE4Editor-AlembicLibrary.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Experimental/CharacterAI/Binaries/Mac/UE4Editor-CharacterAI.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Experimental/HTML5Networking/Binaries/Mac/UE4Editor-HTML5Networking.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/UObjectPlugin/Binaries/Mac/UE4Editor-UObjectPlugin.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/VisualStudioCodeSourceCodeAccess/Binaries/Mac/UE4Editor-VisualStudioCodeSourceCodeAccess.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/SubversionSourceControl/Binaries/Mac/UE4Editor-SubversionSourceControl.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/CLionSourceCodeAccess/Binaries/Mac/UE4Editor-CLionSourceCodeAccess.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/XCodeSourceCodeAccess/Binaries/Mac/UE4Editor-XCodeSourceCodeAccess.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/PerforceSourceControl/Binaries/Mac/UE4Editor-PerforceSourceControl.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Developer/GitSourceControl/Binaries/Mac/UE4Editor-GitSourceControl.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Blendables/LightPropagationVolume/Binaries/Mac/UE4Editor-LightPropagationVolumeRuntime.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Blendables/LightPropagationVolume/Binaries/Mac/UE4Editor-LightPropagationVolumeEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/MovieScene/ActorSequence/Binaries/Mac/UE4Editor-ActorSequence.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/MovieScene/MatineeToLevelSequence/Binaries/Mac/UE4Editor-MatineeToLevelSequence.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/MovieScene/LevelSequenceEditor/Binaries/Mac/UE4Editor-LevelSequenceEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Enterprise/DatasmithContent/Binaries/Mac/UE4Editor-DatasmithContent.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/CableComponent/Binaries/Mac/UE4Editor-CableComponent.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/RuntimePhysXCooking/Binaries/Mac/UE4Editor-RuntimePhysXCooking.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ProceduralMeshComponent/Binaries/Mac/UE4Editor-ProceduralMeshComponent.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ProceduralMeshComponent/Binaries/Mac/UE4Editor-ProceduralMeshComponentEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/Steam/SteamVR/Binaries/Mac/UE4Editor-SteamVR.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/Steam/SteamVR/Binaries/Mac/UE4Editor-SteamVRController.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/AppleARKit/Binaries/Mac/UE4Editor-AppleARKit.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/CustomMeshComponent/Binaries/Mac/UE4Editor-CustomMeshComponent.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/AndroidPermission/Binaries/Mac/UE4Editor-AndroidPermission.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/MobilePatchingUtils/Binaries/Mac/UE4Editor-MobilePatchingUtils.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ImmediatePhysics/Binaries/Mac/UE4Editor-ImmediatePhysics.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ImmediatePhysics/Binaries/Mac/UE4Editor-ImmediatePhysicsEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ExampleDeviceProfileSelector/Binaries/Mac/UE4Editor-ExampleDeviceProfileSelector.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/PhysXVehicles/Binaries/Mac/UE4Editor-PhysXVehicles.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/PhysXVehicles/Binaries/Mac/UE4Editor-PhysXVehiclesEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/AppleMoviePlayer/Binaries/Mac/UE4Editor-AppleMoviePlayer.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/AndroidDeviceProfileSelector/Binaries/Mac/UE4Editor-AndroidDeviceProfileSelector.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/LocationServicesBPLibrary/Binaries/Mac/UE4Editor-LocationServicesBPLibrary.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Runtime/ArchVisCharacter/Binaries/Mac/UE4Editor-ArchVisCharacter.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Animation/LiveLink/Binaries/Mac/UE4Editor-LiveLink.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Animation/LiveLink/Binaries/Mac/UE4Editor-LiveLinkEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Online/OnlineSubsystem/Binaries/Mac/UE4Editor-OnlineSubsystem.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Online/OnlineSubsystemNull/Binaries/Mac/UE4Editor-OnlineSubsystemNull.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Online/OnlineSubsystemUtils/Binaries/Mac/UE4Editor-OnlineSubsystemUtils.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Online/OnlineSubsystemUtils/Binaries/Mac/UE4Editor-OnlineBlueprintSupport.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/2D/Paper2D/Binaries/Mac/UE4Editor-Paper2D.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/2D/Paper2D/Binaries/Mac/UE4Editor-Paper2DEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/2D/Paper2D/Binaries/Mac/UE4Editor-PaperSpriteSheetImporter.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/2D/Paper2D/Binaries/Mac/UE4Editor-PaperTiledImporter.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/2D/Paper2D/Binaries/Mac/UE4Editor-SmartSnapping.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Messaging/TcpMessaging/Binaries/Mac/UE4Editor-TcpMessaging.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Messaging/UdpMessaging/Binaries/Mac/UE4Editor-UdpMessaging.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/FacialAnimation/Binaries/Mac/UE4Editor-FacialAnimation.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/FacialAnimation/Binaries/Mac/UE4Editor-FacialAnimationEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/MobileLauncherProfileWizard/Binaries/Mac/UE4Editor-MobileLauncherProfileWizard.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/AssetManagerEditor/Binaries/Mac/UE4Editor-AssetManagerEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/GameplayTagsEditor/Binaries/Mac/UE4Editor-GameplayTagsEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/CryptoKeys/Binaries/Mac/UE4Editor-CryptoKeys.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/CryptoKeys/Binaries/Mac/UE4Editor-CryptoKeysOpenSSL.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/PluginBrowser/Binaries/Mac/UE4Editor-PluginBrowser.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/SpeedTreeImporter/Binaries/Mac/UE4Editor-SpeedTreeImporter.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Editor/DataValidation/Binaries/Mac/UE4Editor-DataValidation.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AndroidMedia/Binaries/Mac/UE4Editor-AndroidMediaEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AndroidMedia/Binaries/Mac/UE4Editor-AndroidMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/MediaCompositing/Binaries/Mac/UE4Editor-MediaCompositing.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/MediaCompositing/Binaries/Mac/UE4Editor-MediaCompositingEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/MediaPlayerEditor/Binaries/Mac/UE4Editor-MediaPlayerEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/ImgMedia/Binaries/Mac/UE4Editor-ImgMedia.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/ImgMedia/Binaries/Mac/UE4Editor-ImgMediaEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/ImgMedia/Binaries/Mac/UE4Editor-ImgMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/ImgMedia/Binaries/Mac/UE4Editor-ImgMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/ImgMedia/Binaries/Mac/UE4Editor-OpenExrWrapper.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/LinearTimecode/Binaries/Mac/UE4Editor-LinearTimecode.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/WmfMedia/Binaries/Mac/UE4Editor-WmfMediaEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/WmfMedia/Binaries/Mac/UE4Editor-WmfMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/WmfMedia/Binaries/Mac/UE4Editor-WmfMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AvfMedia/Binaries/Mac/UE4Editor-AvfMedia.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AvfMedia/Binaries/Mac/UE4Editor-AvfMediaEditor.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AvfMedia/Binaries/Mac/UE4Editor-AvfMediaFactory.dylib (API version 3944462).
LogModuleManager: Display: Found up-to-date module file ../../../Engine/Plugins/Media/AvfMedia/Binaries/Mac/UE4Editor-AvfMediaFactory.dylib (API version 3944462).
LogInit: Warning: Still incompatible or missing module: UE4Editor-SocketIO.dylib
LogInit: Warning: Still incompatible or missing module: UE4Editor-SocketIOClient.dylib
LogInit: Warning: Still incompatible or missing module: UE4Editor-SIOJson.dylib
LogInit: Warning: Still incompatible or missing module: UE4Editor-CoreUtility.dylib
LogInit: Warning: Still incompatible or missing module: UE4Editor-SIOJEditorPlugin.dylib
LogExit: Preparing to exit.
`

**Comment by anadin** [comment]
I've started to look at the Prebuild project but so far have some CMake errors etc. will get back to it soon

**Comment by getnamo** [comment]
I'm not 100% certain this is the issue but ```was built for newer OSX version (10.12) than being linked (10.11) ``` seems to be a key line from that log, have you tried updating macOS to 10.12 or newer and recompiling? From https://simple.wikipedia.org/wiki/MacOS it seems 10.12 is Sierra and the newest is High Sierra (10.13)

Otherwise it seems to spit out a similar error to the linux build (see this comment: https://github.com/getnamo/socketio-client-ue4/issues/70#issuecomment-369579967)

**Comment by anadin** [comment]
Yup I tried all of the options in #70 and no help, also I am building on 10.13. I set the build targets all to 10.12 just in case, but that did not make any difference. My research on that is that it is a warning and the error, the error seems to be specifically "Undefined symbols for architecture x86_64"

Even though I grabbed the newer libsioclient.a I am thinking we may need to also rebuild the boost .a's for x64. Hence my trying to get the Prebuild stuff working - I'll see if I can get some time on it tomorrow.

**Comment by anadin** [comment]
OK I did a rebuild and now Its working. I did also have to install boost but looking OK I think. If you like I can do a PR for the libs

**Comment by getnamo** [comment]
Awesome, if you have a working libs for both boost and sioclient such that it works in a fresh project without additional dependencies by all means open a pr and I'll merge it.

**Comment by anadin** [comment]
Here you. go. https://github.com/getnamo/socketio-client-ue4/pull/75

**Comment by getnamo** [comment]
Couldn't personally verify but merged the PR here: https://github.com/getnamo/socketio-client-ue4/commit/e7427e63d3f7f10311f1cc0edb2ba92ccd3c4451.

It would be great to have another mac user confirm that version latest master/ v0.7.4 works without additional dependencies.

**Comment by tvervest** [comment]
@anadin I'm trying to run the latest master on macos, but I get [the same error message you had before](https://github.com/getnamo/socketio-client-ue4/issues/15#issuecomment-376043031) (I'm also running macos 10.13). [You mentioned](https://github.com/getnamo/socketio-client-ue4/issues/15#issuecomment-376391401) you did a rebuild and got it to work, could you specify _what_ you rebuilt to get it working? I'll try to do the same to see if I get the same result :)

**Comment by iambeeblebrox** [comment]
Not sure if anybody is still interested, but I just got everything working on a Mac running MacOS Mojave (10.14). Basic steps:
- I downloaded the plug sources from here (Github)
- Download the sample from Github
- Whitelisted Mac for module SocketIOClient in SocketIOClient.uplugin (was already whitelisted for the others)
- Loaded the sample projects --> asked to compile plugin --> error
- After some trial-and-error I decided to rebuild the static libraries from: 
https://github.com/getnamo/socketio-client-prebuild
- I had to fiddle around a bit with building the Boost libs (had to modify build.sh script)
- After I managed to build the static libs, copied them over to the appropriate locations in the plugin directory
- Finally I started the Unreal sample client project (as well as running the node.js sample server) and behold .... it works nicely

Not sure if my libs would work for anything below MacOS Mojave though ... but if someone cares to try it out ...

**Comment by iambeeblebrox** [comment]
Oh and now for something completely different .... iOS and Android. Basic use case is for VR/AR applications working together.

**Comment by getnamo** [comment]
If it's working, feel free to make a pull request with the working libs and I'll merge them, then we could have some other mac users test it out and confirm.

**Comment by iambeeblebrox** [comment]
> If it's working, feel free to make a pull request with the working libs and I'll merge them, then we could have some other mac users test it out and confirm.

I'ld love to do this, but I'm not sure how to ... I'm not exactly fluent in all things Git. I know this: https://github.com/getnamo/socketio-client-ue4/issues/15#issuecomment-265720874 refers to it, but that's about all I know. If you can give me some further hints I will see to it.

**Comment by getnamo** [comment]
To make a pull request you fork this repository(consider using the desktop client to simplify things https://desktop.github.com/), change the files/src with your fixes in your fork and then commit them. Once that's done you make a pull request from your fork (can be done here: https://github.com/getnamo/socketio-client-ue4/pulls).

See also https://help.github.com/articles/creating-a-pull-request/ for help

**Comment by iambeeblebrox** [comment]
Ok. I have just created a pull request: #107

**Comment by getnamo** [comment]
That was a clean pull request 👍, merged https://github.com/getnamo/socketio-client-ue4/commit/a47e89c9a51543482baa0b68c4b75085fc693b4e.

Pinging any other mac user to re-pull the master repo and try out the new static libs by @iambeeblebrox. Let me know if mac osx now works for you too!

**Comment by getnamo** [comment]
@iambeeblebrox switched to including the source directly a separate module. This removes the need for platform specific static libs, could you try pulling the latest master and see if it compiles without issue on MacOS?

**Comment by iambeeblebrox** [comment]
@getnamo sorry it took a while. I was away on an extended holiday ;-). I will test the latest version a.s.a.p.

**Comment by iambeeblebrox** [comment]
@getnamo Green Light!!! I have just compiled a test project with the 1.0 version on my Mac. Works like a charm. This is not an exhaustive test mind you but it compiles and runs without issues anyway.

By the way: is there a reason why you limit the 1.0 release to UE 4.22? I tested it with 4.21 (updated the uplugin file) and that seemed to work fine.

If I find the time I will give iOS a go tomorrow.
